### ISSUE TITLE
Error in first line of .R script files when missing function parenthesis

### ISSUE THREAD

**Issue by galactose310** [jimhester/lintr#3038]
When I open a .R file in VS Code (or Codium) some files are linted with an error at 1st line:
```code
Failed to run diagnostics: ! in callr subprocess.
Caused by error in `call[[1L]]`:
! object of type 'symbol' is not subsettable
```

I reported this in REditorSupport/languageserver: https://github.com/REditorSupport/languageserver/issues/723

They made a commit that make this error disappear in some files, but it remains in some. I found that it still happens in files where functions are called at the end without any argument, an called without parenthesis:
```R
data |>
  filter(id > 3) |>
  summary
```

@randy3k reproduced the issue:
```R
> lintr::lint(text = "data |> filter(id > 3) |> summary")
Error in call[[1L]] : object of type 'symbol' is not subsettable
> packageVersion("lintr")
[1] '3.3.0.1'
```

Although this code is correctly interpreted by R (and is accepted in RStudio or Positron), I believe it is bad practice and one should be warned by lintr. Yet if you want to adhere as closely as possible to the syntax variations of R, lintr should not break like this.
